EIPS/eip-5450.md

```diff
@@ -13,67 +13,127 @@ requires: 3540, 3670, 4200, 4750
 
 ## Abstract
 
-Introduce extended validation of code sections to guarantee that stack underflow cannot happen during execution of validated contracts.
+Introduce extended validation of EOF code sections to guarantee that stack underflow and overflow cannot happen during execution of validated contracts.
 
 ## Motivation
 
-Currently existing EVM implementations perform a number of validity checks for each executed instruction, such as check for stack overflow/underflow, sufficient gas, etc. This change aims to minimize the number of such checks required at run-time, by verifying at deploy-time that no exceptional conditions can happen, and preventing to deploy the code where it could happen.
+The current EVM performs a number of validity checks for each executed instruction, such as checking
+for instruction being defined, stack overflow and underflow, and enough amount of gas remaining.
+
+This EIP minimizes the number of such checks required at run-time
+by verifying that no exceptional conditions can happen
+and preventing the execution and deployment of any invalid code.
+
+The operand stack validation provides several benefits:
+
+- removes run-time stack underflow check for all instructions,
+- removes run-time stack overflow check for all instruction except `CALLF`,
+- ensures that an execution terminates with one of the terminating instructions,
+- prevents the deployment of code with unreachable instructions, which discourages the use of code sections for data storage (TODO),
+- function type information is only needed for validation.
```

Comment:
This means only the input, output info? This might be a little confusing if we also add `maxStackHeight` to the type info.

Reply:
I think actually output also will be needed for "stack cleaning" version of RETF

Reply:
This has been removed because no longer true (e.g. `CALLF` requires `max_stack_height`).